### PR TITLE
Add `icicle` mode and default setting via Preferences.jl

### LINKED ISSUE #22: flip y axis?
@one-more-minute showed me flame plots with the stack growing downward and it was shockingly more intuitive. Perhaps this should be the default?

### PR DESCRIPTION
**Pull request by IanButterworth**
Adds `ProfileView.view(icicle = true)`

Closes #22 ?

Screen Shot 2022-01-16 at 11 42 38 AM

### CODE REVIEW

**Comment by IanButterworth** [comment]
Or `graphmode = :icicle`?

Also, some people might want this as the default?

**Comment by IanButterworth** [comment]
Also, it could be implemented in FlameGraphs instead, and save the `reverse` operations?

**Comment by timholy** [comment]
We could set this up as a Preferences.jl option.

Not that I suspect it matters from a performance standpoint, but you could probably also save the `reverse` by setting a [canvas coordinate system](https://juliagizmos.github.io/GtkObservables.jl/stable/drawing/) with reversed y-axis. I think the orientation is more intrinsically a visual question than an algorithmic one, so I think putting this option here rather than FlameGraphs makes sense (but I can be persuaded otherwise).

**Comment by IanButterworth** [comment]
I thought about reversing the canvas like that, but thought it might also flip the text?

Happy to give it a try.

**Comment by IanButterworth** [comment]
Ok, that all works nicely! Thanks for the tips

File: README.md [hunk 1]
```diff
@@ -195,6 +195,14 @@ Here is the meaning of the different arguments:
 
 - `fontsize` controls the size of the font displayed as a tooltip.
 
+- `expand_threads` controls whether a page is created for each thread (requires julia 1.8, enabled by default)
+
+- `expand_tasks` controls whether a page is shown for each task (requires julia 1.9, off by default)
```
Comment (timholy) [inline README.md:200]: 1.9?

File: test/runtests.jl [hunk 2]
```diff
@@ -47,6 +47,15 @@ end
         @test isa(ProfileView.view(C=true), ProfileView.Window)
         @test isa(ProfileView.view(fontsize=18), ProfileView.Window)
         @test isa(ProfileView.view(windowname="ProfileWindow"), ProfileView.Window)
+        @test isa(ProfileView.view(graphtype=:icicle), ProfileView.Window)
+
+        @test_logs (:info, "Default graphtype set to \"icicle\"") ProfileView.set_graphtype!(:icicle)
+        @test isa(ProfileView.view(), ProfileView.Window)
+        @test_logs (:info, "Default graphtype set to \"flame\"") ProfileView.set_graphtype!(:flame)
+        @test isa(ProfileView.view(), ProfileView.Window)
```
Comment (timholy) [inline test/runtests.jl:55]: If someone runs these lines locally who has their personal default set to `:icicle`, will this reset their default to `:flame`? If so, let's `try/finally` restore whatever setting the user has.

**Review by timholy — APPROVED** [review]
Small tweaks, then merge & release when ready.

File: test/runtests.jl (hunk 2 above)
Comment (IanButterworth) [reply test/runtests.jl]: Good catch. Thanks

**Pull request merged**